Enable pylint rules and fix exposed bugs.

mattclay opened the pull request:
##### SUMMARY

Enable pylint rules and fix exposed bugs:

* invalid-unary-operand-type
* raising-format-tuple
* stop-iteration-return

##### ISSUE TYPE

Bugfix Pull Request

##### COMPONENT NAME

various

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (pylint-bug-fixes a2fd4f2444) last updated 2018/10/17 10:06:27 (GMT -700)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/matt/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/matt/code/mattclay/ansible/lib/ansible
  executable location = /home/matt/code/mattclay/ansible/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

mattclay committed:
Resolve invalid-unary-operand-type.

mattclay committed:
Resolve raising-format-tuple.

mattclay committed:
Resolve stop-iteration-return.

sivel commented on lib/ansible/module_utils/compat/ipaddress.py:
```diff
@@ -346,7 +346,10 @@ def _find_address_range(addresses):
+    try:
+        first = last = next(it)
+    except StopIteration:
+        return
```
Since this is in a vendored file, should we really be updating it like this?

We have history of applying upstream patches to our vendored libs, but not modifying them to diverge.

mattclay replied:
Good catch. This is only a bug when calling `_find_address_range` directly, which is probably why it isn't fixed upstream. Only `collapse_addresses` calls it, and never with an empty list, which is needed to trigger the bug on Python 3.7 and later.

If we're OK with telling people not to call `_find_address_range` directly we could just add an ignore to `test/sanity/pylint/ignore.txt` with a comment stating that we don't intend to fix the issue. All the other ignores listed are intended to be fixed at some point.

@sivel @abadger What do you think?

abadger replied:
I agree we shouldn't make a change to the logic of the code.  I'm okay with whichever methods of ignoring the issue in this file that other people like, whether an ignore.txt or a pylint disable comment or etc

mattclay replied:
I've reverted the code changes in the file and added an inline pylint disable comment. This seemed more appropriate than using the ignore.txt file since we have no intention of changing the logic in the code.

mattclay committed:
Use disable comment instead of fixing logic.

The affected line in _find_address_range will only fail on Python 3.7
and later if the function is called with an empty address list. As an
internal method it is never called in this way, making it a non-issue
for use via public methods.

Using a comment to disable the rule in favor of an ignore.txt entry
since there are no plans to change the logic in the code itself. This
will also prevent any potential future issues being added in other
parts of the code when updating it based on upstream changes.

abadger commented:
+1
